frontend/model/contracts/group.js

```diff
@@ -1242,13 +1291,32 @@ sbp('chelonia/defineContract', {
           updateGroupStreaks({ state, getters })
           getters.groupSettings.distributionDate = period
         }
+      },
+      sideEffect ({ meta, contractID }, { getters }) {
+        const { loggedIn } = sbp('state/vuex/state')
+        const myProfile = getters.groupProfile(loggedIn.username)
+
+        if (isActionYoungerThanUser(meta, myProfile) &&
+          myProfile.incomeDetailsType // if the user has entered their income details
+        ) {
+          sbp('gi.notifications/emit', 'NEW_DISTRIBUTION_PERIOD', {
+            creator: meta.username,
+            createdDate: meta.createdDate,
+            groupID: contractID,
+            memberType: myProfile.incomeDetailsType === 'pledgeAmount' ? 'pledger' : 'receiver'
+          })
+        }
```

Comment:
The fact that we're in a new distribution period can be noticed and emitted locally without touching the contracts code